clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -807,10 +816,10 @@ protected void close(Timer timer) {
      * Leave the current group and reset local generation/memberId.
      */
     public synchronized void maybeLeaveGroup() {
-        if (!coordinatorUnknown() && state != MemberState.UNJOINED && generation.isValid()) {
+        if (!coordinatorUnknown() && state != MemberState.UNJOINED && generation.hasMemberId()) {
```

Comment:
How much effort would it be to have a test case for this? We have a few  LeaveGroup tests in `ConsumerCoordinatorTest`.

Reply:
this has been added. resolving comment.